discord/bot.py

```diff
@@ -298,17 +295,15 @@ def _check_command(cmd: ApplicationCommand, match: Mapping[str, Any]) -> bool:
                         falsy_vals = (False, [])
                         for opt in value:
                             cmd_vals = (
-                                [val.get(opt, MISSING) for val in as_dict[check]]
+                                [val.get(opt) for val in as_dict[check]]
```

Comment:
Neither `as_dict` nor `to_check` will ever contain `cog` in this chunk of the code. I don't understand how this fixes the issue, MISSING was again used for matching commands.

Reply:
I’ll need to test this more. Earlier when I had *accidentally* inverted the logic, it fixed the issue with non-cog commands, but now when properly treating None as a drop in replacement for MISSING it's again not retrieving options.